cylc/flow/run_modes/dummy.py

```diff
@@ -0,0 +1,125 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utilities supporting dummy mode.
+
+Dummy mode shares settings with simulation mode.
+"""
+
+from typing import TYPE_CHECKING, Any, Dict, Tuple
+
+from cylc.flow.run_modes.simulation import (
+    ModeSettings,
+    disable_platforms,
+    get_simulated_run_len,
+    parse_fail_cycle_points
+)
+from cylc.flow.run_modes import RunMode
+from cylc.flow.platforms import get_platform
+
+
+if TYPE_CHECKING:
+    from cylc.flow.task_job_mgr import TaskJobManager
+    from cylc.flow.task_proxy import TaskProxy
+    from typing_extensions import Literal
+
+
+CLEAR_THESE_SCRIPTS = [
+    'init-script',
+    'env-script',
+    'pre-script',
+    'post-script',
+    'err-script',
+    'exit-script',
+]
+
+
+def submit_task_job(
+    task_job_mgr: 'TaskJobManager',
+    itask: 'TaskProxy',
+    rtconfig: Dict[str, Any],
+    workflow: str,
+    now: Tuple[float, str]
+) -> 'Literal[False]':
+    """Submit a task in dummy mode.
+
+    Returns:
+        False indicating that TaskJobManager needs to continue running the
+        live mode path.
+    """
+    configure_dummy_mode(
+        rtconfig, itask.tdef.rtconfig['simulation']['fail cycle points'])
+
+    itask.summary['started_time'] = now[0]
+    task_job_mgr._set_retry_timers(itask, rtconfig)
+
+    itask.mode_settings = ModeSettings(
+        itask,
+        task_job_mgr.workflow_db_mgr,
+        rtconfig
+    )
+
+    itask.platform = get_platform()
+    itask.platform['name'] = RunMode.DUMMY.value
```

Comment:
Can modifying the platform cause changes to other platform dicts elsewhere in the system.

I.E. is the dict returned by `get_platform` guaranteed to be a unique object (`copy.deepcopy`)?

Reply:
YES. https://github.com/cylc/cylc-flow/blob/01bbca591da4d405e8f8ea8cf9302e4d15e3df59/cylc/flow/platforms.py#L250C13-L250C26

Lots of potential problems otherwise.

Reply:
That deepcopy only runs if the platform name matches a regex?

Reply:
All Platforms must match that regex as that's how we're selecting platforms. If it doesn't match any of the platform definition regexes then it's not  a defined platform.